arrows-stdlib-cats/src/main/scala/arrows/stdlib/cats.scala

```diff
@@ -0,0 +1,127 @@
+package arrows.stdlib
+
+import cats.{ Applicative, StackSafeMonad }
+import cats.effect._
+import cats.arrow.ArrowChoice
+import cats.kernel.{ Monoid, Semigroup }
+import cats.mtl.ApplicativeAsk
+
+import scala.concurrent.{ ExecutionContext, Promise }
+import scala.util.control.NonFatal
+import scala.util.{ Failure, Success }
+
+object Cats extends ArrowInstances
+
+sealed abstract class ArrowInstances extends ArrowInstances1 {
+
+  private final object ToLeft {
```

Comment:
I took a look at the bytecode and it doesn't seem that `final object` will allow constant folding by the JIT compiler. Sorry for the back and forth, but could you change it to:

```scala
private final val toLeftInstance: Any => Left[Any, Nothing] = Left(_)
private final def toLeftI[T] = instance.asInstanceOf[T => Left[T, Nothing]]
```